addons/website/tests/test_performance.py

```diff
@@ -27,10 +50,39 @@ def _get_url_hot_query(self, url, cache=True):
         # ensure worker is in hot state
         self.url_open(url)
         self.url_open(url)
+        self.url_open(url)
+
+        sql_count_before = self.registry.test_cr.sql_log_count
+
+        if table_count:
+            self.registry.test_cr.sql_track_table_count = True
+            sql_from_log_before = copy.deepcopy(self.registry.test_cr.sql_from_log)
+            sql_into_log_before = copy.deepcopy(self.registry.test_cr.sql_into_log)
 
-        sql_count = self.registry.test_cr.sql_log_count
         self.url_open(url)
-        return self.registry.test_cr.sql_log_count - sql_count - EXTRA_REQUEST
+
+        sql_count = self.registry.test_cr.sql_log_count - sql_count_before - EXTRA_REQUEST
+        if table_count:
+            self.registry.test_cr.sql_track_table_count = False
```

Comment:
In case of exception above, this line will not be executed.
A context manager would be much better in this case:
```py
with (self.cr._enable_table_tracking() if table_count else nullcontext()):
    ...
```